lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -975,6 +1001,91 @@ def configure_network(self, vm_obj):
                 self.configspec.deviceChange.append(nic)
                 self.change_detected = True
 
+    def configure_vapp_properties(self, vm_obj):
+        if len(self.params['vapp_properties']) == 0:
+            return
+
+        for x in self.params['vapp_properties']:
+            if not x.get('id'):
+                self.module.fail_json(msg="id is required to set vApp property")
+
+        is_vapp_changed = False
+
+        new_vmconfig_spec = vim.vApp.VmConfigSpec()
+
+        # This is primarily for vcsim/integration tests, unset vAppConfig was not seen on my deployments
+        orig_spec = vm_obj.config.vAppConfig if vm_obj.config.vAppConfig else new_vmconfig_spec
+
+        vapp_properties_current = dict((x.id, x) for x in orig_spec.property)
+        vapp_properties_to_change = dict((x['id'], x) for x in self.params['vapp_properties'])
+
+        # each property must have a unique key
+        # init key counter with max value + 1
+        all_keys = [x.key for x in orig_spec.property]
+        new_property_index = max(all_keys) + 1 if all_keys else 0
+
+        for property_id, property_spec in vapp_properties_to_change.items():
+
```

Comment:
is_vapp_changed = False

Reply:
The flag should be set to True if _any_ property changed, so It can't be inside this cycle.  Otherwise. if no changes to the last property in the list, the vm update will not be triggered. Will move it to 1026.

Reply:
I think you didn't understand what I was suggesting.

What you do: Add each of the property the user have asked for in new_vmconfig_spec.property, and if at least one of them modify the properties already attached to the VM → apply new_vmconfig_spec
What I propose: Only add to new_vmconfig_spec.property the property the user have asked for that actually modify (add/remove/edit) the properties already attached to the VM. And then, if there is some properties in new_vmconfig_spec.property → apply new_vmconfig_spec.

Reply:
I think you didn't understand what I was suggesting.

What you do
: Add each of the property the user have asked for in new_vmconfig_spec.property, and if at least one of them modify the properties already attached to the VM → apply new_vmconfig_spec

What I propose
: Only add to new_vmconfig_spec.property the property the user have asked for that actually modify (add/remove/edit) the properties already attached to the VM. And then, if there is some properties in new_vmconfig_spec.property → apply new_vmconfig_spec.

Reply:
Thanks, that makes sense now. Now collecting only properties that needs update, and if the list is not empty, assigning vApp configuration.